orthopairs/src/main/java/org/reactome/release/orthopairs/Main.java

```diff
@@ -0,0 +1,155 @@
+     *  3) With the species Gene-Protein file from Biomart and the Gene Homolog file from Compara, map out a Protein homolog file
+     *
+     *  In this updated version, all of this information comes straight from 2 flat files provided by PANTHER. This makes things much quicker and stabler.
+     *  The only twist to this though, is that not all of the species Gene IDs come from Ensembl, but rather there is some that come from organism-specific databases.
+     *  The solution has been to use alternative ID mapping files from these organism-specific databases to convert any non-Ensembl IDs to their Ensembl equivalent.
+     *
+     *  At the end, we have the same two files for each species: The Gene-Protein mapping file specific to the species, and the Protein Homolog file.
+     *  All Protein IDs are UniProt (instead of the mix of Ensembl and UniProt) and all Gene IDs match the format used by Ensembl.
+     *
+     *  These files are much smaller than the older version, since PANTHER annotates which homologs are the 'Least Diverged', meaning we can filter out many that add noise to the dataset.
+     *
+     */
+
+    private static final Logger logger = LogManager.getLogger();
+
+    public static void main( String[] args ) throws IOException, ParseException {
+
+        String pathToConfig = "";
+        String sourceMappingSpecies = "";
+        // If using an alternative source species, specify the 4-letter code as the second argument
+        //TODO: Better solution for this. What if user wants default config but to change source species?
+        if (args.length > 0) {
+            pathToConfig = args[0];
+            sourceMappingSpecies = args[1];
+        } else {
+            pathToConfig = "src/main/resources/config.properties";
+            sourceMappingSpecies = "hsap";
+        }
+        Properties props = new Properties();
+        props.load(new FileInputStream(pathToConfig));
+        // Load all config properties
+        String releaseNumber = props.get("releaseNumber").toString();
+        String pathToSpeciesConfig = props.get("pathToSpeciesConfig").toString();
+        String pantherFilepath = props.get("pantherFileURL").toString();
+        String pantherQfOFilename = props.get("pantherQfOFilename").toString();
+        String pantherHCOPFilename = props.get("pantherHCOPFilename").toString();
+        String MGIFileURL = props.get("MGIFileURL").toString();
+        String RGDFileURL = props.get("RGDFileURL").toString();
+        String XenbaseFileURL = props.get("XenbaseFileURL").toString();
+        String ZFINFileURL = props.get("ZFINFileURL").toString();
+
+        new File(releaseNumber).mkdir();
+
+        logger.info("Starting Orthopairs file generation");
+        // Download and extract homology files from Panther
+        List<String> pantherFiles = new ArrayList<String>(Arrays.asList(pantherQfOFilename, pantherHCOPFilename));
```

Comment:
You can use the diamond operator when instantiating a parameterized object.

`List<String> pantherFiles = new ArrayList<>(...);`

Reply:
Yea I typically do that now, that one just got through

Reply:
I think IntelliJ will color it grey if it finds the type on the right side and let you know it can be removed.

You could try it out by pasting in something like this to see if it works.

`List<String> a = new ArrayList<String>();`